torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -64,9 +64,9 @@ auto AccumulateGrad::apply(variable_list&& grads) -> variable_list {
     // a thing never promised and documented, but used in some hacks seen
     // on the internet.
     if (grad_variable.is_sparse() && !new_grad.is_sparse()) {
-      grad_variable.set_data(new_grad.data() + grad_variable.data());
+      grad_variable = new_grad + grad_variable;
     } else {
-      grad_variable.data() += new_grad.data();
+      grad_variable.set_data(grad_variable + new_grad);
```

Comment:
does it not work to just do `grad_variable += new_grad;` ?

Reply:
@gchanan I can confirm that this does in-place update to `grad_variable`, both based on https://github.com/pytorch/pytorch/blob/810816a1f95e4bbbd9b7416d317875aecc5f40a5/aten/src/ATen/TensorOperators.h#L24-L26 and this test case:
```python
params = torch.Tensor([1.5, 1.5]).requires_grad_()
y = torch.FloatTensor([1.]).requires_grad_()[0]
with torch.no_grad():
    params.grad = torch.Tensor([1.5, 1.5])

loss = (params[1] - params[0] ** 2)
grad_saved = params.grad
loss.backward()   # This will hit `grad_variable += new_grad;`
assert id(grad_saved) == id(params.grad)
```